### ISSUE TITLE
AnimatedNavHost with ModalBottomSheetLayout Error

### ISSUE THREAD

**Issue by Jparrgam** [raamcosta/compose-destinations#229]
Hello, I have a problem with the `1.7.16-beta` version generates an exception like this 

```
java.lang.IllegalStateException: popBackStack was called with androidx.navigation.NavBackStackEntry@3405775b which does not exist in back stack [androidx.navigation.NavBackStackEntry@df66c224, androidx.navigation.NavBackStackEntry@3405775b]
        at androidx.navigation.Navigator.popBackStack(Navigator.kt:186)
        at androidx.navigation.NavController.popBackStackInternal(NavController.kt:274)
        at androidx.navigation.NavController.popBackStackInternal(NavController.kt:557)
        at androidx.navigation.NavController.popBackStackInternal$default(NavController.kt:515)
        at androidx.navigation.NavController.setGraph(NavController.kt:1083)
        at androidx.navigation.NavController.setGraph(NavController.kt:100)
        at com.google.accompanist.navigation.animation.AnimatedNavHostKt.AnimatedNavHost(AnimatedNavHost.kt:146)
        at com.google.accompanist.navigation.animation.AnimatedNavHostKt$AnimatedNavHost$11.invoke(Unknown Source:23)
        at com.google.accompanist.navigation.animation.AnimatedNavHostKt$AnimatedNavHost$11.invoke(Unknown Source:10)
```

I'm using the compose version `1.3.0-alpha02` compose compiler `1.3.0-rc01` 

My code

```
val navHostEngine = rememberAnimatedNavHostEngine(
            navHostContentAlignment = Alignment.TopCenter,
            rootDefaultAnimations = RootNavGraphDefaultAnimations.ACCOMPANIST_FADING,
            defaultAnimationsForNestedNavGraph = mapOf(
                DashboardRoutersConfig to NestedNavGraphDefaultAnimations(
                    enterTransition = { fadeIn(animationSpec = tween(2000)) },
                    exitTransition = { fadeOut(animationSpec = tween(2000)) }
                ),
            )
        )

        val navController = navHostEngine.rememberNavController()

        val bottomSheetNavigator = rememberBottomSheetNavigator()
        navController.navigatorProvider += bottomSheetNavigator

        CompositionLocalProvider(
            LocalDependencyContainer
                    provides dependencyContainer
        ) {
            ModalBottomSheetLayout(
                bottomSheetNavigator,
                sheetShape = RoundedCornerShape(topEnd = 16.dp, topStart = 16.dp),
            ) {
                DestinationsNavHost(
                    navController = navController,
                    navGraph = Root,
                    startRoute = if (destination == Screens.SplashScreen.route) {
                        Root.startRoute
                    } else {
                        LoginNavGraph
                    },
                    engine = navHostEngine,
                    dependenciesContainerBuilder = {
                        dependency(
                            LoginAndSignUpNavigator(navController, activity)
                        )
                        dependency(activity)
                        dependency(application)
                    }
                )
            }
        }
```

**Comment by Jparrgam** [comment]
@raamcosta Has this ever happened to you?

**Comment by raamcosta** [comment]
Hi @Jparrgam !
Sorry I haven’t gotten back to you yet regarding this. I’ve been super tied with time these past days.

This never happened to me, and at least at the moment I don’t see a problem with your code. I also doubt this can be related specifically with my library, it’s probably some misusage of compose navigation in general.

Did this code ever work for you before? Also can you try to make it simpler and see if somewhere along the line it starts working? It may give us a clue as to what exactly is causing it.

thanks a lot!

**Comment by Jparrgam** [comment]
Hi @raamcosta, sorry to bother you, it was a version and compatibility error.

**Comment by raamcosta** [comment]
Hi @Jparrgam can you please tell me what version combination was not working for you?

**Comment by jakhongirmadaminov** [comment]
I have faced the same exception after upgrading to 
`1.7.22-beta` with 
```
compose = "1.3.0-rc01"
composeCompiler = "1.3.2"
```
fixed by downgrading `1.7.22-beta` to `1.5.14-beta` with
```
compose = "1.3.0-rc01"
composeCompiler = "1.3.2"
```
